spec/base_decorator_spec.rb

```diff
@@ -0,0 +1,19 @@
+require 'spec_helper'
+
+require_relative '../base_decorator'
+
+RSpec.describe BaseDecorator do
+  describe '#correct_name' do
+    it 'delegates the correct_name to the decorated object' do
+      nameable = double('Nameable')
+      allow(nameable).to receive(:correct_name).and_return('John Doe')
+
+      decorator = BaseDecorator.new(nameable)
+
+      result = decorator.correct_name
+
+      expect(nameable).to have_received(:correct_name)
+      expect(result).to eq('John Doe')
+    end
+  end
```

Comment:
- [ ] @Team great work adding tests for the base decorator class, however, you forgot to add tests for the `initialize` method, kindly consider adding tests for this method.